### ISSUE TITLE
Feature request: Color lines to indicate which node they originate from

### ISSUE THREAD

**Issue by akinnee** [sverweij/dependency-cruiser#469]
### Context

With a bunch of same-colored lines, it's hard to follow where they go.

### Expected Behavior

Each node could have a unique outline color and the edges originating from it  can match that same color.

### Current Behavior

All nodes and edges are the same color.

### Possible Solution

- When a node is created, pick a random color or generate a stable color by hashing the node's text.
- Set the outline color of that node to the color.
- Arrows pointing outward from the node should have the same color.

**Comment by sverweij** [comment]
Hi @akinnee I’ve bumped into this as well and it’s why these two features in dependency-cruiser exist
- with [theming](https://github.com/sverweij/dependency-cruiser/blob/develop/doc/options-reference.md#theme-dot-ddot-and-archi-reporters) you can colour modules and dependencies that match regular expressions 
- [depcruise-wrap-stream-in-html](https://github.com/sverweij/dependency-cruiser/blob/develop/doc/cli.md#depcruise-wrap-stream-in-html) transforms diagrams so they have highlight-on-hover.


The [real world samples](https://github.com/sverweij/dependency-cruiser/blob/develop/doc/real-world-samples.md) page has several examples that employ both techniques.

Do these features meet your need?

**Comment by akinnee** [comment]
@sverweij, thanks! If I set the color of a module, will the edges coming from it match that color?

**Comment by sverweij** [comment]
@akinnee currently I've only used it the other way around (the line gets the color of the dependency it's pointing _to_) - but I understand the need to use the colors on the _from_ side. And maybe even both (with a gradient? will have to see how usable that'd be, though)

I'm looking into your suggestion and see if I can make it simpler so you don't have to spec out modules individually (or even by group/ regex) i.e. by providing something that automatically colors from a pallet. I'm trying to prototype it first to see how usable that'd be - but got a little lost in color theory. I'll dump an update here.
